include/envoy/network/listen_socket.h

```diff
@@ -90,6 +90,39 @@ class ConnectionSocket {
    * Close the underlying socket.
    */
   virtual void close() PURE;
+
+  /**
+   * Visitor class for setting socket options.
+   */
+  class Options {
+  public:
+    virtual ~Options() {}
+
+    /**
+     * @param socket the socket on which to apply options.
+     * @return true if succeeded, false otherwise.
+     */
+    virtual bool setOptions(ConnectionSocket& socket) const PURE;
+
+    /**
+     * @return bits that can be used to separate upstream connections based on the options. Should
```

Comment:
nit: remove "upstream". The concept applies to any connection I think.

Reply:
Removed.